Training5/manageapp/src/redux/reducer/userReducer.js

```diff
@@ -0,0 +1,23 @@
+import { initialState } from "../state/state";
+
+export default function reducer(state = initialState, action) {
+    switch (action.type) {
+        case "SAGA_GET_ALL_USERS_INFO":
```

Comment:
@Huynhkimphat, You don't need to separate action using on the saga, it simple is action type, so you need to declare all actions in ActionNames.js

Reply:
and don't need to add prefix SAGA_, this is redundant

Reply:
you missing handle take error or loading, so we don't have these information to display to the UI